packages/dds/tree/src/simple-tree/schemaFactoryRecursive.ts

```diff
@@ -192,11 +40,57 @@ export function createFieldSchemaUnsafe<
+ *
+ * The non-recursive versions of the schema building methods will run into several issues when used recursively.
+ * Consider the following example:
  *
- * @remarks This is separated from SchemaFactory as these APIs are more experimental, and may be stabilized independently.
+ * ```typescript
+ * const Test = sf.array(Test); // Bad
+ * ```
  *
- * @sealed @internal
+ * This has several issues:
+ *
+ * 1. It is a structurally named schema.
+ * Structurally named schema derive their name from the names of their child types, which is not possible when the type is recursive since its name would include itself.
+ * Instead a name must be explicitly provided.
+ *
+ * 2. The schema accesses itself before its defined.
+ * This would be a runtime error if the TypeScript compiler allowed it.
+ * This can be fixed by wrapping the type in a function, which also requires explicitly listing the allowed types an an array (`[() => Test]`)
+ *
+ * 3. TypeScript fails to infer the recursive type, and fall back to any with the warning or error (depending on the compiler configuration):
+ * `'Test' implicitly has type 'any' because it does not have a type annotation and is referenced directly or indirectly in its own initializer.ts(7022)`.
+ * This issue is what the specialized recursive schema building methods fix.
+ * This fix comes at a cost: to make the recursive cases work, the `extends` clauses had to be removed.
+ * This means that mistakes declaring recursive schema often don't give compile errors in the schema.
+ * Additionally support for implicit construction had to be disabled.
+ * This means that new nested {@link Unhydrated} nodes can not be created like `new Test([[]])`.
+ * Instead the nested nodes must be created explicitly using the construction like`new Test([new Test([])])`.
+ *
+ * 4. It is using "POJO" mode since its not explicitly declaring a new class.
+ * This means that if the schema generated d.ts files replace recursive references with `any`, breaking use of recursive schema across compilation boundaries.
+ * This is fixed by explicitly creating a class which extends the returned schema.
+ *
+ * All together, the fixed version looks like:
+ * ```typescript
+ * class Test extends sf.arrayRecursive("Test", [() => Test]) {} // Good
+ * ```
+ *
+ * Be very careful when declaring recursive schema.
+ * Due to the removed extends clauses, subtly mistake will compile just fine, but cause strange errors when the schema is used.
+ *
+ * For example if the square branches around the allowed types are forgotten:
+ *
+ * ```typescript
+ * class Test extends sf.arrayRecursive("Test", () => Test) {} // Bad
+ * ```
+ * This schema will still compile, and some (but not all) usages of it may look like they work correctly while other usages will produce generally unintelligible compile errors.
+ * This issue can be partially mitigated using{@link ValidateRecursiveSchema}.
+ * @sealed @beta
```

Comment:
Probably worth adding a `@see {@link ValidateRecursiveSchema}` here, since the reference to it in the above description comes buried rather late in the text. Would add an extra helpful link in the generated docs.